test/express-route.test.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var path = require('path'),
+  helpers = require('yeoman-generator').test,
+  assert = require('yeoman-generator').assert,
+  temp = require('temp').track();
+
+describe('Express Sub Generators Tests', function () {
+  this.timeout(0);
+  /**
+   * Setup the temp directory
+   */
+  before(function (done) {
+    helpers.testDirectory(path.join(__dirname, 'temp'), done);
+  });
+
+  /**
+   * Clean up temp directory
+   */
+  after(function () {
+    temp.cleanup();
+  });
+
+  describe('Generate an express route through the sub-generator', function () {
+    beforeEach(function (done) {
+      helpers.run(path.join(__dirname, '../express-route'))
+        .withOptions({
+          'skip-install': true
+        })
+        .withArguments(['foo'])
+        .withPrompts({
+          'moduleName': 'core'
+        })
+        .on('ready', function (generator) {
+          // this is called right before `generator.run()` is called
+        })
+        .on('end', function () {
+          done();
+        });
+    });
+
+    it('should generate an express route file', function () {
+      assert.file('modules/core/server/routes/foo.server.routes.js');
```

Comment:
I will be devil's advocate as well:
at the same time, here is the example, where having multiple controllers/tests/models etc. in one model might make sense structure-wise.
Suppose, you have "clothes" module. It would be an ok design to group under it both controller/model/routes for "clothes", but also for subcategories of closes, like "pants", "t-shirts", "socks".

If there are not too many subcategories, then merging all of them under the same folder kinda makes sense.